File: Daily/FairCandySwap.java
import java.util.*;

/**
 *  leetcode 888
 */
public class FairCandySwap {
    public int[] fairCandySwap(int[] A, int[] B) {
        // 求到目前A和B的数量
         int sumA = Arrays.stream(A).sum();
         int sumB = Arrays.stream(B).sum();
        //  sumA-x+y = sumB-y+x   =>     x = y+(sumA-sumB)/2
         int count = (sumA-sumB)/2;
         // 通过hash来判断
         HashSet<Integer> set = new HashSet<Integer>();
         for (int i : A) {
             set.add(i);
             
         }
         int []ans = new int[2];
         for(int y:B){
             int x = y+count;
             if(set.contains(x)){
                 ans[0] = x;
                 ans[1] = y;
                 break;
             }
         }
         return ans;

    }}

File: Daily/FairCandySwap.py
from typing import List
class Test:
    def fairCandySwap(self, A: List[int], B: List[int]) -> List[int]:
       # 遍历两个数字求的合
       sumA,sumB = sum(A),sum(B)
       # 得到需要的中间值
       count = (sumA-sumB)//2
       setA = set(A)
       ans = None
       for y in B:
           x = y+count
           if x in setA:
               ans = [x,y]
               break
       return ans


